crates/agentgateway/src/http/ext_authz.rs

```
@@ -42,62 +102,248 @@ impl ExtAuthz {
 			target: self.target.clone(),
 			client,
 		};
-		let mut client = AuthorizationClient::new(chan);
-		let tcp_info = req.extensions().get::<TCPConnectionInfo>().unwrap();
-		let tls_info = req.extensions().get::<TLSConnectionInfo>();
+		let mut grpc_client = AuthorizationClient::new(chan);
+		// Get connection info with proper error handling
+		// Clone the fields we need to avoid borrow checker issues
+		let (peer_addr, local_addr) = {
+			let tcp_info = req.extensions().get::<TCPConnectionInfo>().ok_or_else(|| {
+				warn!("TCPConnectionInfo not found in request extensions");
+				ProxyError::Processing(anyhow::anyhow!("Missing TCP connection info"))
+			})?;
+			(tcp_info.peer_addr, tcp_info.local_addr)
+		};
+		let tls_info = req.extensions().get::<TLSConnectionInfo>().cloned();
 
-		// Convert headers to the format expected by Envoy
+		// Convert headers to the format expected by Envoy, applying allowlist if configured
+		// Handle multi-value headers by joining with comma (HTTP spec)
+		// Limit header count to prevent DoS attacks
+		const MAX_HEADERS: usize = 100;
```

Not sure this belongs here; if we have a header limit we probably want it at a higher level and reject the request entirely (this is what Envoy does IIUC). Silently dropping headers can result in an authorization bypass